src/core/impl/TChem_Impl_SingleParticleAerosolWater.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef __TCHEM_IMPL_SINGLEPARTICLEAEROSOLWATER_HPP__
+#define __TCHEM_IMPL_SINGLEPARTICLEAEROSOLWATER_HPP__
+
+#include "TChem_Util.hpp"
+
+namespace TChem {
+namespace Impl {
+
+  template<typename ValueType, typename DeviceType>
+struct AerosolWater_SingleParticle
+{
+    using value_type = ValueType;
+    using device_type = DeviceType;
+    using scalar_type = typename ats<value_type>::scalar_type;
+
+    using real_type = scalar_type;
+    /// sacado is value type
+    using value_type_1d_view_type = Tines::value_type_1d_view<value_type,device_type>;
+    using real_type_1d_view_type = Tines::value_type_1d_view<real_type,device_type>;
+    using ordinal_type_1d_view_type = Tines::value_type_1d_view<ordinal_type,device_type>;
+    using aerosol_model_data_type= AerosolModel_ConstData<device_type>;
+
+    using const_real_type_1d_view_type = Tines::value_type_1d_view<const real_type,device_type>;
+
+    template<typename MemberType>
+    KOKKOS_INLINE_FUNCTION static
+    void team_invoke(const MemberType& member,
+        ordinal_type i_part,
+        real_type number_conc,
+        real_type state[],  // TODO: figure out how the state vector actually works
+        //const AerosolModelData amd
+        const aerosol_model_data_type& amcd,
+        ordinal_type rh_idx,
+        ordinal_type aqueous_water_idx
+        //const aerosol_ion_pair_type ionpair,
+        //const std::map<std::string, int> ion_idx_map,
+        //const std::map<std::string, real_type> atomic_weights  // TODO: pack into ionpair objects
+        )
+    {   
+        // NOTE: Cannot use std::string with cuda 
+        // Declare and initialize variables for later use
+        
+        ordinal_type anion_idx;
+        ordinal_type cation_idx;
+        ordinal_type ion_species_idx;
+        ordinal_type qty;
+        std::string atom_name;
+        std::string ion_name;
+        real_type ion_molecular_weight;
+        real_type cation;
+        real_type anion;
+        real_type e_ac;
+        real_type e_aa;
+        real_type ALPHA = -100.0;
+        real_type ion_conc;
+        real_type water_content;
+        real_type NW;
+        real_type MW;
+        real_type ZW;
+        real_type MW_H20 = 999.7351; // 55.51*18.01
```

Comment:
We must remove these hard-coded constants from src/core/impl.

Reply:
`NW`, `MW`, and `ZW` are assigned values based on attributes of an `ionpair` object with calculation type `EQSAM`. Should I instead simply use those attribute values in place instead of assigning them to separate local variables?

Reply:
I was referring to` real_type MW_H20 = 999.7351.` Let's move this value to `src/core/TChem_Util.hp`p. Additionally, I suggest adding a postfix to clarify the units of this molecular weight.

For example: see this [line of code](https://github.com/PCLAeroParams/TChem-atm/blob/85a250caab6a04b4a376c48c2d5264760bf0db9c/src/core/TChem_Util.hpp#L200) where we defined a constant value.

Reply:
Also, let's use an enum instead of std::string or char for type.

Reply:
Resolved by moving water molality and molecular weight to `TChem_Util.cpp`